### ISSUE TITLE
Sync master and 2.2 tests and tests API

### ISSUE THREAD

**Issue by pierrejoye** [libgd/libgd#231]
I do like the new functions in gdTests. Makes the whole thing clearer and easier to maintain.

On the bad side, It becomes very painful to work on master and 2.2 from a tests point of view. Let sync them and add what we need to add. :)

Please comment, thoughts, ideas, needs :)

**Comment by vapier** [comment]
we could just stop adding tests to the 2.2 branch ;)

does a cherry pick back not go well ?

**Comment by pierrejoye** [comment]
Merge conflicts all the time.

And we will have to maintain 2.2 for a year or so, for php. I prefer to
have this part synced then follow the:

. commit lower active branch
. merge to higher branch up to master accordingly.

That makes git history easier as well as the maintenance.

**Comment by vapier** [comment]
i said cherry-pick, not merge.  imo, you should use cherry-pick most of the time instead of merge.  the merge command just makes the history noisy and a pain to follow.

```
$ git checkout GD-2.2
$ git cherry-pick d6c50766ee39d96f8b9899acd6e0d358623b9812
<fix conflicts>
$ git add <files>
$ git cherry-pick --continue
``

**Comment by pierrejoye** [comment]
It makes sense too. We can try after 2.2.2.

I would still like to make both 2.2 and master more in line to ease bugs
fixes in both branches. :)

It will also help to ensure 2.3 does not introduce BC breaks.

**Comment by vapier** [comment]
i've cherry picked back the fix to the GD-2.2 branch now
